Modules/_collectionsmodule.c

```diff
@@ -1706,6 +1868,7 @@ dequeiter_next(dequeiterobject *it)
         it->b = it->b->rightlink;
         it->index = 0;
     }
+    deque_unlock(it->deque);
     Py_INCREF(item);
```

Comment:
This incref should happen inside the lock, no?  Otherwise there is a possible race (and GC), right?